pkg/utils/types.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+package utils
+
+import (
+	"errors"
+	"math"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+var (
+	digitIndicates = []string{
+		"...",
+		"..",
+		".",
+		"-",
+		" ",
+	}
+	formatIndicate   = ";"
+	variableIndicate = "."
+)
+
+type Attribute struct {
+	Describe    string // [attribute(b 64, b-64, b..64)]; [format(MMDD, hhmmss)]
+	Description string
+}
+
+func (s Attribute) ElementType() (*ElementType, error) {
+	attribute := s.Describe
+	for _, indicate := range digitIndicates {
+		if strings.Contains(attribute, indicate) {
+			var format string
+			if strings.Contains(attribute, formatIndicate) {
+				splits := strings.Split(attribute, formatIndicate)
+				attribute = splits[0]
+				format = strings.TrimSpace(splits[len(splits)-1])
```

Comment:
We need a length check here. `strings.Split` will always return at least one item, but could return only one.